### DISCUSSION TITLE
How can I indirectly refer two models?

### DISCUSSION THREAD

**Discussion by ugrky** [prisma/prisma#4880] [Q&A]
Hello community <3

I am new to Prisma, and exploring my possibilities with it. There is something I am trying to figure out how to achieve, which is:

I have a `Product`, which has different `Variation`s, which has `VariationValue`s. I allow user's to subscribe into specific `VariationValue` of a product, and want to be able to get a `subscription` <-> `product` relation within Prisma level. Does this model seem fair, or could I re-structure it in a smarter way?

```prisma
model Product {
  id           Int    @id @default(autoincrement())
  name         String
  description  String
  sku          String
  defaultPrice Int

  // relation fields
  variations     Variation[]
}

model Variation {
  id   Int    @id @default(autoincrement())
  name String

  // relation fields
  lootBoxId Int
  lootBox   LootBox          @relation(fields: [lootBoxId], references: [id])
  values    VariationValue[]
}

model VariationValue {
  id    Int    @id @default(autoincrement())
  name  String
  sku   String
  price Int

  variationId Int
  variation   Variation @relation(fields: [variationId], references: [id])
}

model Subscription {
  id               Int                   @id @default(autoincrement())
  state            SubscriptionState     @default(ACTIVE)
  subscriptionDate DateTime              @default(now())

  // relation fields
  userId Int
  user   User @relation(fields: [userId], references: [id])

  variationValueId Int
  variationValue VariationValue @relation(fields: [variationValueId], references: [id])
}
```

**Comment by ryands17** [comment]
Hey @ugrky 👋 

If your `Subscription` is related directly to the `VariationValue` of the `Product`, you can fetch the `Product` from the `Subscription` as follows:

```ts
  await db.subscription.findUnique({
    where: { id: 1 },
    include: {
      variationValue: {
        include: { variation: { include: { Product: true } } },
      },
    },
  })
```

Also your Prisma schema would need a back relation to `Product` from the `Variation` table as follows:

```prisma
model Product {
  id           Int    @id @default(autoincrement())
  name         String
  description  String
  sku          String
  defaultPrice Int

  // relation fields
  variations Variation[]
}

model Variation {
  id   Int    @id @default(autoincrement())
  name String

  // relation fields
  values    VariationValue[]
  Product   Product?         @relation(fields: [productId], references: [id])
  productId Int?
}

model VariationValue {
  id    Int    @id @default(autoincrement())
  name  String
  sku   String
  price Int

  variationId  Int
  variation    Variation      @relation(fields: [variationId], references: [id])
  Subscription Subscription[]
}

model Subscription {
  id               Int      @id @default(autoincrement())
  subscriptionDate DateTime @default(now())

  // relation fields
  variationValueId Int
  variationValue   VariationValue @relation(fields: [variationValueId], references: [id])
}
```

Let me know if this helps :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
